open_instruct/finetune.py

```diff
@@ -95,6 +95,13 @@ class FlatArguments:
         default="main",
         metadata={"help": "The specific model version to use (can be a branch name, tag name or commit id)."},
     )
+    chat_template_file: Optional[str] = field(
+        default=None,
+        metadata={"help": (
+            "A file with its content as a jinja2 template for encoding chat messages. "
+            "If not provided, we will use the default Tulu template."
+        )}
+    )
```

Comment:
I love the ability to specify custom chat template. What do you think about doing something like below? We should be able to unify them in the future, but I didn't want to do a massive breaking change given the current sprint.

https://github.com/allenai/open-instruct/blob/de62ff4e8fc183e9158c2e57914b60b16c84c0fc/open_instruct/dataset_processor.py#L88-L136

https://github.com/allenai/open-instruct/blob/de62ff4e8fc183e9158c2e57914b60b16c84c0fc/open_instruct/dataset_processor.py#L147


https://github.com/allenai/open-instruct/blob/de62ff4e8fc183e9158c2e57914b60b16c84c0fc/open_instruct/reward_modeling.py#L238

Reply:
Yeah, this should work well. 

Now I realize there are multiple places where we rely on the chat template (sft, dpo, rm training, rs), and they all have their different functions/collators to do the encoding. Shall we unify them? Probably need your help to do this then @vwxyzjn

Reply:
That's what the [dataset_processor.py](https://github.com/allenai/open-instruct/blob/de62ff4e8fc183e9158c2e57914b60b16c84c0fc/open_instruct/dataset_processor.py#L88-L136) is for :) we can unify them post tulu 3.

Maybe for now what do you think about a simple fix like importing the `CHAT_TEMPLATE` from `open_instruct/dataset_processor.py`? 


If we do the unification now, I am worried it might introduce subtle bugs / regresssions, hindering the current project.

Reply:
Fair, I will keep this change within the SFT part. Let me modify a bit to use these templates.